### DISCUSSION TITLE
Poco Build fail for openssl

### DISCUSSION THREAD

**Discussion by deepanshug** [pocoproject/poco#4159] [Q&A]
Hi, 

I am trying to install POCO library on windows by running "buildwin.cmd 170" command but it fails C:\delete\poco\Crypto\include\Poco\Crypto\Crypto.h(37,10): fatal  error C1083: Cannot open include file: 'openssl/opens
slv.h': No such file or directory (compiling source file src\RSAKey.cpp) [C:\delete\poco\Crypto\Crypto_vs170.vcxproj].

I have tried so far:
Downloaded precompiled binaries of openssl and put it in C:/delete/poco folder and executed command after but it fails.

Can you please guide me here.

**Comment by aleks-f** [comment]
You can start by looking at what the error says - compiler can't find headers, not binaries. Binaries come into play at link time, but you first have to have headers, and compiler must know where to find them.

**Reply by deepanshug** [reply]
I am trying to compile POCO using the buidwin script provided. Error is specific to OpenSSL missing headers.
- I have downloaded OpenSSL and placed in POCO_BASE
- Environment variable POCO_BASE is set
- "Components" file updated to include ÖpenSSL during compilation.

I could see the build is not respecting the components file or environment variable. Still I am in the same phase where I could not build POCO on Windows.
